commit a5bd8fc2e2a1f2a1daa3be32092ae9cd2c62e972
Author: Robert Bogart <robert.bogart@joyent.com>
Date:   2019-04-19T02:33:56+00:00 (6 months ago)
    
    MANTA-4233 Accelerated garbage collection omits account information from instructions
    Reviewed by: Elijah Zupancic <elijah.zupancic@joyent.com>
    Reviewed by: Richard Kiene <richard.kiene@joyent.com>
    Approved by: Richard Kiene <richard.kiene@joyent.com>

diff --git a/lib/delete_record_transformer.js b/lib/delete_record_transformer.js
index 7d6475c..3d08549 100644
--- a/lib/delete_record_transformer.js
+++ b/lib/delete_record_transformer.js
@@ -156,7 +156,20 @@ _process_record(record, done)
 	var key = record.key;
 	var sharks = [];
 
-	var line_data = [value.creator, value.objectId];
+	/*
+	 * Reminder: As discussed earlier, object metadata for objects created
+	 * with the Manta multipart upload API do not have creator uuids.  In
+	 * such a case, we can obtain the uuid from the `owner' field instead.
+	 * Failure to do so will prevent us from deleting the object later on
+	 * down the road because we will not have a means of associating the
+	 * object with an account (a necessary requirement for object deletion
+	 * on a mako).  On a more positive note, this is the last stop in the
+	 * chain where it is necessary to dig through a record for this kind of
+	 * information.  From here on out, account information will be passed
+	 * forward via `line_data'.
+	 */
+	var creator = value.creator || value.owner;
+	var line_data = [creator, value.objectId];
 
 	if (value.type !== 'object' || value.contentLength === 0) {
 		done(null, null, key);
